src/main/scala/org/broadinstitute/dsde/workbench/avram/api/CollectionsServlet.scala

```diff
@@ -1,39 +1,28 @@
 package org.broadinstitute.dsde.workbench.avram.api
 
-import io.circe.generic.auto._
-import io.circe.syntax._
 import java.util.UUID
-import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
 
-import org.broadinstitute.dsde.workbench.avram.model.{AvramException, SamResource}
+import io.circe.generic.auto._
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.broadinstitute.dsde.workbench.avram.model.SamResource
 import org.broadinstitute.dsde.workbench.avram.service.CollectionsService
 
 
 class CollectionsServlet extends HttpServlet with AvramServlet {
 
   private val collectionsService = new CollectionsService()
 
-  override def doGet(request: HttpServletRequest, response: HttpServletResponse) = {
-    val collection = handleAuthenticatedRequest(request, response) { userInfo =>
-      response.setContentType("application/json")
-      val parameter = request.getPathInfo.stripPrefix("/")
-      val collectionExternalId = UUID.fromString(parameter)
-      val future = collectionsService.getCollection(collectionExternalId)
-      registerOnComplete(future, request, response)
+  override def doGet(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+    handleAuthenticatedRequest(request, response) { _ =>
+      val collectionExternalId = UUID.fromString(request.getPathInfo.stripPrefix("/"))
+      collectionsService.getCollection(collectionExternalId)
     }
-    response.getWriter.write(collection.asJson.noSpaces)
   }
 
-  override def doPost(request: HttpServletRequest, response: HttpServletResponse) = {
-    val thing = handleAuthenticatedRequest(request, response) { userInfo =>
-      response.setContentType("application/json")
-      val parameter = request.getPathInfo.stripPrefix("/")
-      val samResource = SamResource(parameter)
-      val future = collectionsService.createCollection(samResource, userInfo.userEmail)
-      registerOnComplete(future, request, response)
+  override def doPost(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+    handleAuthenticatedRequest(request, response) { userInfo =>
+      val samResource = SamResource(request.getPathInfo.stripPrefix("/"))
```

Comment:
Same as above -- `request.getPathInfo` can return `null` if there is no path. This should probably with `Either`.